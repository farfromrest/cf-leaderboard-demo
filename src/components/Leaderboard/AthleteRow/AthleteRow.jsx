import React from 'react'
import numeral from 'numeral'
import { arrayOf, shape, string } from 'prop-types'

import style from './AthleteRow.scss'

function Rank (props) {
  return numeral(props.children).format('0o')
}

function AthleteRow (props) {
  const { rank, picture, name, region, points, workouts } = props

  return (
    <div className={style.wrapper}>
      <div className={style.rank}>
        <div className={style.value}>{rank}</div>
      </div>
      <div className={style.picture}>
        <img src={picture} />
      </div>
      <div className={style.name}>{name}</div>
      <div className={style.region}>{region}</div>
      <div className={style.points}>{points}</div>
      {workouts.map((workout, index) => (
        <div className={style.workout} key={index}>
          <Rank>{workout.rank}</Rank>
          <div className={style.score}>&nbsp;({workout.score})</div>
        </div>
      ))}
    </div>
  )
}

AthleteRow.propTypes = {
  rank: string,
  picture: string,
  name: string,
  region: string,
  points: string,
  workouts: arrayOf(
    shape({
      rank: string,
      score: string
    })
  )
}

AthleteRow.defaultProps = {
  workouts: []
}

export default AthleteRow
