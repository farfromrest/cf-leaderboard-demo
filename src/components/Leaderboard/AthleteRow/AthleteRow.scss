@import 'colors';

.wrapper {
  background-color: $white;
  border: 1px solid $gray;
  border-radius: 10px;
  margin-bottom: 10px;
  display: grid;
  grid-template-columns: 80px 80px 30% repeat(6, 1fr);
  grid-template-rows: repeat(2, 1fr);
  // grid-template-areas:
  //   'rank pic name points score1 score2 score3 score4 score5'
  //   'rank pic region points score1 score2 score3 score4 score5';

  &:nth-child(odd) {
    background-color: darken($white, 3%);
  }

  .rank {
    grid-column: 1;
    grid-row: 1 / span 2;
    font-weight: 600;
    font-size: 2em;
    border-right: 1px solid $gray;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .picture {
    grid-column: 2;
    grid-row: 1 / span 2;
    width: 50px;
    height: 50px;
    border-radius: 50%;
    overflow: hidden;
    align-self: center;
    justify-self: center;
    border: 1px solid $gray;

    img {
      max-width: 100%;
    }
  }

  .name {
    grid-column: 3;
    grid-row: 1;
    align-self: end;
    font-weight: 600;
    font-size: 1.3em;
  }

  .region {
    grid-column: 3;
    grid-row: 2;
    padding-top: 5px;
    color: darken($gray, 10%);
  }

  .workout,
  .points {
    border-left: 1px solid lighten($gray, 10%);
    align-self: center;
    text-align: center;
    font-size: 1em;
    font-weight: 500;
    line-height: 4.5;
    grid-row: 1 / span 2;
  }

  .score {
    color: darken($gray, 10%);
    display: inline-block;
  }
}
