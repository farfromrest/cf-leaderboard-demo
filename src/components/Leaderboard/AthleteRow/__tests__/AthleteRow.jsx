import React from 'react'
import renderer from 'react-test-renderer'

import AthleteRow from '../AthleteRow'

it('renders correctly', () => {
  const tree = renderer.create(<AthleteRow />).toJSON()
  expect(tree).toMatchSnapshot()
})
