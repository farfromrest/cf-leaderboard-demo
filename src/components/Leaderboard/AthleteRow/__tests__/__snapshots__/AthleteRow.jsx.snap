// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="rank"
  >
    <div
      className="value"
    />
  </div>
  <div
    className="picture"
  >
    <img
      src={undefined}
    />
  </div>
  <div
    className="name"
  />
  <div
    className="region"
  />
  <div
    className="points"
  />
</div>
`;
